src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -30,6 +31,28 @@ public partial class CSharpCompilation
         /// </summary>
         private Symbol[] _lazyWellKnownTypeMembers;
 
+        private bool _needsGeneratedIsReadOnlyAttribute_IsFrozen;
+        private bool _needsGeneratedIsReadOnlyAttribute_Value;
```

Comment:
Do we need this state in both `Compilation` and `PEAssemblyBuilder`? Would it be sufficient to keep state here only?